Runtime/Generated/ReactAction.cs

```diff
@@ -0,0 +1,605 @@
+// ------------------------------------------------------------------------------
+// <auto-generated>
+//     This code was generated by a tool.
+//
+//     Changes to this file may cause incorrect behavior and will be lost if
+//     the code is regenerated.
+// </auto-generated>
+// ------------------------------------------------------------------------------>
+using System;
+using ReactUnity.Helpers;
+
+namespace ReactUnity.Helpers
+{
+  /// <summary>
+  ///   A wrapper class for the <see cref="ReactUnity.Helpers.Callback" /> class to provide convienient access to the
+  ///   to delegates and events in ReactUnity.
+  /// </summary>
+  class ReactAction
+  {
+      /// <summary>
+      ///   The underlying action.
+      /// </summary>
+      private Action _delegate { get; set; }
+
+      /// <summary>
+      ///   Adds the listener to the action, and returns a function that removes the listener. This method should be only called from React.
+      /// </summary>
+      [Obsolete("This method should be only called from React.", true)]
```

Comment:
I think this should only be a warning. Also, the message should discourage the usage from C#, but not completely prohibit it.

Reply:
I wanted to keep the standard syntax of the delegates as much as possible with += and -=. That's why I wanted to discourage AddListener usage from C#.

If JS engine can access private methods, we can change this to internal to remove the Obsolete warning/error

Reply:
Yes but `AddListener` returns a handy "RemoveListener" method, which could be useful also for C# users. `+` and `-` operators are prone to errors as you did with the example.

Reply:
Before I push this, do you think we should have then 2 methods with the same signature?
```
public Action AddListener(object callback) // For React with Obsolete
public Action AddListener(Action callback) // For C#
```

or should we merge them into one, and resolve inside of it if it's Action or Callback?

Reply:
I think another signature for C# is a good idea.

Reply:
I think it should also be `AddListener` because we already use `AddListener` in several places.